src/Campaigns/Actions/RedirectDisabledCampaignPage.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Give\Campaigns\Actions;
+
+use Give\Campaigns\Models\CampaignPage;
+
+/**
+ * @unreleased
+ */
+class RedirectDisabledCampaignPage
+{
+    /**
+     * @unreleased
+     */
+    public function __invoke()
+    {
+        if('give_campaign_page' !== get_post_type()) {
+            return;
+        }
+
+        $campaignPage = CampaignPage::find(get_the_ID());
+
+        if(!$campaignPage->campaign()->enableCampaignPage) {
+            global $wp_query;
+            $wp_query->set_404();
+            status_header(404);
+            get_template_part(404);
+            exit;
```

Comment:
```suggestion
```

We can remove the exit statement so WP can still render a page, otherwise the website is unreachable.  

<img width="1512" alt="Screenshot 2025-02-07 at 3 43 53 PM" src="https://github.com/user-attachments/assets/a6825087-d889-440f-9c6f-ad204d250d6e" />

vs 

<img width="1510" alt="Screenshot 2025-02-07 at 3 44 07 PM" src="https://github.com/user-attachments/assets/cc7dc99f-cff4-4abc-b9a2-dfb54c7c3e5a" />

Also, it looks like we don't need the `get_template_part` either as WP must handle that up the chain via `get_404_template()`

Reply:
Ok, interesting. This didn't cause any issues for my with a default theme.